apps/prairielearn/src/pages/administratorSettings/administratorSettings.ts

```diff
@@ -41,6 +44,15 @@ router.post(
       }
       const jobSequenceId = await chunks.generateAllChunksForCourseList(course_ids, authn_user_id);
       res.redirect(res.locals.urlPrefix + '/administrator/jobSequence/' + jobSequenceId);
+    } else if (req.body.__action === 'sync_context_documents' && isEnterprise()) {
+      const client = new OpenAI({
+        apiKey: config.openAiApiKey ? config.openAiApiKey : undefined,
```

Comment:
What's going on here?

Reply:
The API key parameter is of type `string | undefined`, while `config.openAiApiKey` is of type `string | null`, so the linter was causing problems without the conversion

Reply:
```suggestion
        apiKey: config.openAiApiKey ?? undefined,
```

Reply:
It wouldn't do us any good to try to run this code if `openAiApiKey` is nullish, as we know any requests downstream of this will fail. You probably want this little snippet we've used elsewhere:

```ts
    if (!config.openAiApiKey || !config.openAiOrganization) {
      throw new error.HttpStatusError(403, 'Not implemented (feature not available)');
    }
```

Reply:
Changed in bd95296541cf8f04550cd7c3e806f3f321fb9d7f.